lib/enum/enum.rb

```diff
@@ -29,6 +28,20 @@ def delete_all
         super
       end
 
+      private
+
+      def define_enum_methods
+        return if @enum_accessors.blank?
+
+        @records&.each do |record|
+          enum_vallue = @enum_accessors.map { |name| record.attributes[name] }.join("_").downcase.gsub(/\W+/, '_').gsub(/^_|_$/, '')
```

Comment:
Did this subexpression come from somewhere else? Why not use something like Active Support's `#camelize` instead of this?

Also, how is this implementation going to handle collisions (e.g., two values like "FooBar" and "foo_bar" that would create the same method name)? How does rails handle collisions? I would like to see some test coverage for a collision case as well

Reply:
The enum_value was derived from this section.
https://github.com/active-hash/active_hash/blob/a192385350b8abc11b77937e3556ad9f3fde587e/lib/enum/enum.rb#L48-L49

> Also, how is this implementation going to handle collisions (e.g., two values like "FooBar" and "foo_bar" that would create the same method name)?

I try it. 
A DuplicateEnumAccessor error occurs when the set_constant method is executed.

https://github.com/active-hash/active_hash/blob/a192385350b8abc11b77937e3556ad9f3fde587e/lib/enum/enum.rb#L39

And We have also added a test for cases where the method is already defined by itself.
https://github.com/active-hash/active_hash/pull/321/commits/00c149770bedfb7d5798dc2c44b65349ca8c93c6